internal/clients/project.go

```diff
@@ -76,7 +84,14 @@ func SetProjectNameWithRetry(client *Client, params *project_service.ProjectServ
 	op := func() error {
 		var err error
 		res, err = client.Project.ProjectServiceSetName(params, nil)
-		return err
+		if err != nil {
+			return err
+		}
+		// Wait for the project name to be set, if an operation ID is returned.
+		if res.Payload.OperationID != "" {
+			return waitForProjectOperation(params.Context, client, "set project name", params.ID, res.Payload.OperationID)
+		}
+		return nil
```

Comment:
is the noop expectd here? if yes, it would be easier to read with a early return.